### PR TITLE
Add trace for SIMPLIFY_ALGEBRA matches

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Adding trace for a specific set of a matchers for simplify algebra here

If we use the default MIGRAPHX_TRACE_MATCHES this significantly slows down debugging efforts when debugging modules and associated matches

Added overloaded call to the find_matches() set of functions to handle this. Arisen from debugging issues with reshape/simplify_algebra.

Most likely can be applied for other modules that perform matches as well by adding the trace argument to find_matches

related #1822

### CODE REVIEW

File: src/include/migraphx/matcher.hpp [hunk 1]
```diff
@@ -428,6 +428,62 @@ void find_matches(Mod& mod, Ms&&... ms)
     }
 }
 
+/// Find matches for an instruction in the module for per section of matchers
+template <class Mod, class... Ms>
+void find_matches(int trace_mod, Mod& mod, instruction_ref ins, Ms&&... ms)
```
Comment (umangyadav) [inline src/include/migraphx/matcher.hpp:433]: You can keep just one overload and make line 427 do `find_matches(0, mod, ins, ms...);`

Comment (TedThemistokleous) [reply src/include/migraphx/matcher.hpp]: Done

File: src/include/migraphx/matcher.hpp [hunk 2]
```diff
@@ -372,9 +372,9 @@ match::matcher_result find_match(module& modl, M&& m)
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_TRACE_MATCHES)
 MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_VALIDATE_MATCHES)
 
-/// Find matches for an instruction in the module
+/// Find matches for an instruction in the module for per section of matchers
 template <class Mod, class... Ms>
-void find_matches(Mod& mod, instruction_ref ins, Ms&&... ms)
+void find_matches(size_t trace_mod, Mod& mod, instruction_ref ins, Ms&&... ms)
```
Comment (umangyadav) [inline src/include/migraphx/matcher.hpp:377]:
```suggestion
void find_matches(size_t trace_pass, Mod& mod, instruction_ref ins, Ms&&... ms)
```
IMO, trace_pass would be better name. `MIGRAPHX_TRACE_MATCHES` is also already matching inside the module.

Comment (TedThemistokleous) [reply src/include/migraphx/matcher.hpp]: Done

**Review by umangyadav — APPROVED** [review]

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**